src/masternodes/mn_rpc.cpp

```diff
@@ -3768,6 +3769,213 @@ UniValue sendtokenstoaddress(const JSONRPCRequest& request) {
 
 }
 
+static UniValue getcustomtx(const JSONRPCRequest& request)
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    RPCHelpMan{"getcustomtx",
+        "\nGet detailed information about a DeFiChain custom transaction. Will search wallet transactions and mempool transaction,\n"
+        "if a blockhash is provided and that block is available then details for that transaction can be returned. -txindex\n"
+        "can be enabled to return details for any transaction.",
+        {
+            {"txid", RPCArg::Type::STR, RPCArg::Optional::NO, "The transaction id"},
+            {"blockhash", RPCArg::Type::STR_HEX, RPCArg::Optional::OMITTED_NAMED_ARG, "The block in which to look for the transaction"},
+        },
+        RPCResult{
+            "{\n"
+            "  \"type\":               (string) The transaction type.\n"
+            "  \"valid\"               (bool) Whether the transaction was valid.\n"
+            "  \"results\"             (json object) Set of results related to the transaction type\n"
+            "  \"block height\"        (string) The block height containing the transaction.\n"
+            "  \"blockhash\"           (string) The block hash containing the transaction.\n"
+            "  \"confirmations\": n,   (numeric) The number of confirmations for the transaction."
+            "}\n"
+        },
+        RPCExamples{
+            HelpExampleCli("getcustomtx", "\"66ea2ac081e2917f075e2cca7c1c0baa12fb85c469f34561185fa64d7d2f9305\"")
+                    + HelpExampleRpc("getcustomtx", "\"66ea2ac081e2917f075e2cca7c1c0baa12fb85c469f34561185fa64d7d2f9305\"")
+        },
+    }.Check(request);
+
+    const uint256 hash(ParseHashV(request.params[0], "txid"));
+
+    CTransactionRef tx;
+    uint256 hashBlock;
+
+    // Search wallet if available
+    if (pwallet) {
+        LOCK(pwallet->cs_wallet);
+        auto it = pwallet->mapWallet.find(hash);
+        if (it != pwallet->mapWallet.end())
+        {
+            tx = it->second.tx;
+            hashBlock = it->second.hashBlock;
+        }
+    }
+
+    CBlockIndex* blockindex{nullptr};
+
+    // No wallet or not a wallet TX, try mempool, txindex and a block if hash provided
+    if (!pwallet || !tx)
+    {
+        if (!request.params[1].isNull()) {
```

Comment:
Should check the request.params size before accessing the element.

Reply:
No need, this is typical behaviour across Bitcoin codebase to check for a defined optional argument.